src/phoenix/trace/span_evaluations.py

```diff
@@ -0,0 +1,73 @@
+import pandas as pd
+
+EVALUATIONS_INDEX_NAME = "context.span_id"
+RESULTS_COLUMN_NAMES = ["score", "label", "explanation"]
+
+EVAL_NAME_COLUMN_PREFIX = "eval."
+
+
+class SpanEvaluations:
+    """
+    SpanEvaluations is a set of evaluation annotations for a set of spans.
+    SpanEvaluations encompasses the evaluation annotations for a single evaluation task
+    such as toxicity or hallucinations.
+
+    SpanEvaluations can be appended to TraceDatasets so that the spans and
+    evaluations can be joined and analyzed together.
+
+    Parameters
+    __________
+    eval_name: str
+        the name of the evaluation, e.x. 'toxicity'
+    dataframe: pandas.DataFrame
+        the pandas dataframe containing the evaluation annotations Each row
+        represents the evaluations on a span.
+
+    Example
+    _______
+
+    DataFrame of evaluations for toxicity may look like:
+
+    | span_id | score              | label              | explanation        |
+    |---------|--------------------|--------------------|--------------------|
+    | span_1  | 1                  | toxic              | bad language       |
+    | span_2  | 0                  | non-toxic          | violence           |
+    | span_3  | 1                  | toxic              | discrimination     |
+    """
+
+    dataframe: pd.DataFrame
+
+    eval_name: str  # The name for the evaluation, e.x. 'toxicity'
+
+    def __init__(self, eval_name: str, dataframe: pd.DataFrame):
+        self.eval_name = eval_name
+
+        # If the dataframe contains the index column, set the index to that column
+        if EVALUATIONS_INDEX_NAME in dataframe.columns:
+            dataframe.set_index(EVALUATIONS_INDEX_NAME, inplace=True)
+
+        # validate that the dataframe is indexed by context.span_id
+        if dataframe.index.name != EVALUATIONS_INDEX_NAME:
+            raise ValueError(
+                f"The dataframe index must be '{EVALUATIONS_INDEX_NAME}' but was "
+                f"'{dataframe.index.name}'"
+            )
+
+        # Drop the unnecessary columns
+        extra_column_names = dataframe.columns.difference(RESULTS_COLUMN_NAMES)
+        dataframe.drop(extra_column_names, axis=1)
+        self.dataframe = dataframe
```

Comment:
drop returns a copy. the pandas modus operandi to write everything like a math equation
```suggestion        
        self.dataframe = dataframe.drop(extra_column_names, axis=1)
```

Reply:
It doesn't I think. It caused my tests to fail:
```python
@overload
    def drop(
        self,
        labels: IndexLabel = ...,
        *,
        axis: Axis = ...,
        index: IndexLabel = ...,
        columns: IndexLabel = ...,
        level: Level = ...,
        inplace: Literal[True],
        errors: IgnoreRaise = ...,
    ) -> None:
```

Reply:
it does a return a copy in practice

<img width="283" alt="Screenshot 2023-12-01 at 12 47 26 PM" src="https://github.com/Arize-ai/phoenix/assets/80478925/b635edcd-3c60-4490-8f8a-f5b85a291715">